encoding/ccf/encode.go

```diff
@@ -0,0 +1,1766 @@
+}
+
+// encodeUInt128 encodes cadence.UInt128 as
+// language=CDDL
+// uint128-value = bigint .ge 0
+func (e *Encoder) encodeUInt128(v cadence.UInt128) error {
+	return e.enc.EncodeBigInt(v.Big())
+}
+
+// encodeUInt256 encodes cadence.UInt256 as
+// language=CDDL
+// uint256-value = bigint .ge 0
+func (e *Encoder) encodeUInt256(v cadence.UInt256) error {
+	return e.enc.EncodeBigInt(v.Big())
+}
+
+// encodeWord8 encodes cadence.Word8 as
+// language=CDDL
+// word8-value = uint .le 255
+func (e *Encoder) encodeWord8(v cadence.Word8) error {
+	return e.enc.EncodeUint8(uint8(v))
+}
+
+// encodeWord16 encodes cadence.Word16 as
+// language=CDDL
+// word16-value = uint .le 65535
+func (e *Encoder) encodeWord16(v cadence.Word16) error {
+	return e.enc.EncodeUint16(uint16(v))
+}
+
+// encodeWord32 encodes cadence.Word32 as
+// language=CDDL
+// word32-value = uint .le 4294967295
+func (e *Encoder) encodeWord32(v cadence.Word32) error {
+	return e.enc.EncodeUint32(uint32(v))
+}
+
+// encodeWord64 encodes cadence.Word64 as
+// language=CDDL
+// word64-value = uint .le 18446744073709551615
+func (e *Encoder) encodeWord64(v cadence.Word64) error {
+	return e.enc.EncodeUint64(uint64(v))
+}
+
+// encodeFix64 encodes cadence.Fix64 as
+// language=CDDL
+// fix64-value = (int .ge -9223372036854775808) .le 9223372036854775807
+func (e *Encoder) encodeFix64(v cadence.Fix64) error {
+	return e.enc.EncodeInt64(int64(v))
+}
+
+// encodeUFix64 encodes cadence.UFix64 as
+// language=CDDL
+// ufix64-value = uint .le 18446744073709551615
+func (e *Encoder) encodeUFix64(v cadence.UFix64) error {
+	return e.enc.EncodeUint64(uint64(v))
+}
+
+// encodeArray encodes cadence.Array as
+// language=CDDL
+// array-value = [* value]
+func (e *Encoder) encodeArray(v cadence.Array, tids ccfTypeIDByCadenceType) error {
+	staticElementType := v.ArrayType.Element()
+
+	// Encode array head with number of array elements.
+	err := e.enc.EncodeArrayHead(uint64(len(v.Values)))
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(v.Values); i++ {
+		// Encode element as value.
+		err = e.encodeValue(v.Values[i], staticElementType, tids)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// encodeDictionary encodes cadence.Dictionary as
+// language=CDDL
+// dict-value = [* (key: value, value: value)]
+func (e *Encoder) encodeDictionary(v cadence.Dictionary, tids ccfTypeIDByCadenceType) error {
+	if len(v.Pairs) > 1 {
+		return e.encodeSortedDictionary(v, tids)
+	}
+
+	dictionaryType := v.DictionaryType.(*cadence.DictionaryType)
+	if dictionaryType == nil {
+		return fmt.Errorf("unexpected dictionary type %s", v.DictionaryType.ID())
+	}
+
+	staticKeyType := dictionaryType.KeyType
+	staticElementType := dictionaryType.ElementType
+
+	// Encode array head with array size of 2 * number of pairs.
+	err := e.enc.EncodeArrayHead(uint64(len(v.Pairs) * 2))
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(v.Pairs); i++ {
+		pair := v.Pairs[i]
```

Comment:
```suggestion
	for _, pair := range v.Pairs {
```

Reply:
Done in commit 2ce376a2b6267226e8ae337c5f0f22e5b7a18dc2.